app/services/movie_api/movie_data.rb

```diff
@@ -0,0 +1,8 @@
+class MovieData
+  def initialize(data)
+    data.each do |key, value|
+      instance_variable_set("@#{key}", value)
+      self.class.send(:attr_accessor, key)
```

Comment:
I think this should be changed to :attr_reader so as to limit outside modification.

Reply:
ohh yes definitely